### PR TITLE
added CortexAttackSurfaceManagement to unremovable packs

### PR DESCRIPTION
**Pull request by yucohen**
## Description
added CortexAttackSurfaceManagement to unremovable packs

### CODE REVIEW

**Review by daryakoval — APPROVED** [review]

File: .gitlab/ci/on-push.yml [hunk 1]
```diff
@@ -528,6 +528,7 @@ xsiam_server_ga:
     GCS_MACHINES_BUCKET: "marketplace-v2-dist-dev/upload-flow/builds-xsiam"
     SERVER_TYPE: "XSIAM"
     MARKETPLACE_NAME: "marketplacev2"
+    UNREMOVABLE_PACKS: "Base,CoreAlertFields,Core,CortexAttackSurfaceManagement "
```
Comment (daryakoval) [inline .gitlab/ci/on-push.yml:531]:
```suggestion
    UNREMOVABLE_PACKS: "Base,CoreAlertFields,Core,CortexAttackSurfaceManagement"
```

File: Tests/scripts/uninstall_packs_and_reset_bucket_cloud.sh [hunk 2]
```diff
@@ -14,5 +14,5 @@ else
   gsutil -m cp -r "gs://$GCS_SOURCE_BUCKET/content" "gs://$GCS_MACHINES_BUCKET/$CLOUD_CHOSEN_MACHINE_ID/" > "$ARTIFACTS_FOLDER/Copy_prod_bucket_to_cloud_machine_cleanup.log" 2>&1
   echo "sleeping 120 seconds"
   sleep 120
-  python3 ./Tests/Marketplace/search_and_uninstall_pack.py --cloud_machine $CLOUD_CHOSEN_MACHINE_ID --cloud_servers_path $CLOUD_SERVERS_PATH --cloud_servers_api_keys "cloud_api_keys.json"
+  python3 ./Tests/Marketplace/search_and_uninstall_pack.py --cloud_machine $CLOUD_CHOSEN_MACHINE_ID --cloud_servers_path $CLOUD_SERVERS_PATH --cloud_servers_api_keys "cloud_api_keys.json" --unremovable_packs $UNREMOVABLE_PACKS
```
Comment (kobymeir) [inline Tests/scripts/uninstall_packs_and_reset_bucket_cloud.sh:17]:
```suggestion
  python3 ./Tests/Marketplace/search_and_uninstall_pack.py --cloud_machine $CLOUD_CHOSEN_MACHINE_ID --cloud_servers_path $CLOUD_SERVERS_PATH --cloud_servers_api_keys "cloud_api_keys.json" --unremovable_packs "${UNREMOVABLE_PACKS}"
```
Prevent globbing and expansion of bash variables

File: Tests/Marketplace/search_and_uninstall_pack.py [hunk 3]
```diff
@@ -124,12 +126,12 @@ def wait_for_uninstallation_to_complete(client: demisto_client):
...
     sleep_duration = 150
     try:
-        installed_packs = get_all_installed_packs(client)
+        installed_packs = get_all_installed_packs(client, unremovable_packs)
         # Monitoring when uninstall packs don't work
         installed_packs_amount_history, failed_uninstall_attempt_count = len(installed_packs), 0
         # new calculation for num of retries
         retries = math.ceil(len(installed_packs) / 2)
-        while len(installed_packs) > len(UNREMOVABLE_PACKS):
+        while len(installed_packs) > len(unremovable_packs):
```
Comment (kobymeir) [inline Tests/Marketplace/search_and_uninstall_pack.py:134]: i think set operation here is preferred:
```
while len(installed_packs - unremoveable_packs)
```
As we check explicitly that whatever is left is only the unremovable.

File: Tests/Marketplace/Tests/search_and_unistall_pack_test.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ def test_exception_reset_base_pack(def_call, return_val, mocker):
    """
     client = MockClient()
     client.api_client.configuration.host = 'https://api-someurl.com/'  # disable-secrets-detection
-    ret_val_from_call = def_call(client)
+    ret_val_from_call = def_call(client, ['Base'])
```
Comment (kobymeir) [inline Tests/Marketplace/Tests/search_and_unistall_pack_test.py:115]: more the `['Base']` list to be a parameter for the test, and test for more scenarios: Empty, 1 pack, 2 packs.

**Review by kobymeir — CHANGES_REQUESTED** [review]

**Pull request merged**